pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +423,51 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+// AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
+	config := NewNonPersistentDisk(datastoreURI)
+	d, err := m.CreateAndAttach(op, config)
```

Comment:
@jzt There's no code checking if the disk is already attached, and I do not see any reference counting/increment in this path.
Please work with @caglar10ur to make sure that reference counting is actually used and that we successfully handle the case where the disk is already attached - this will become the common path if we add lazy detach to support the containerStatPath function.
This should also apply to mounting of NFS volumes as well.